In [1]:
# Import the Important Libraries.
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
import plotly.express as px
import scipy.stats as st 


In [2]:
# Store all the Data in the Pandas DataFrame.
Food = pd.read_excel(r"D:\DATA ANALYST\Swiggy\Raw Dataset\food.xlsx")
Menu = pd.read_excel(r"D:\DATA ANALYST\Swiggy\Raw Dataset\menu.xlsx")
Orders = pd.read_excel(r"D:\DATA ANALYST\Swiggy\Raw Dataset\orders.xlsx")
Orders_Type = pd.read_excel(r"D:\DATA ANALYST\Swiggy\Raw Dataset\orders_type.xlsx")
Restaurant = pd.read_excel(r"D:\DATA ANALYST\Swiggy\Raw Dataset\restaurant.xlsx")
Users = pd.read_excel(r"D:\DATA ANALYST\Swiggy\Raw Dataset\users.xlsx")

**Initial Data Exploration**


In [3]:
# Print the first 5 rows.
Food.head()

f_id                  item veg_or_non_veg
0  fd0     Aloo Tikki Burger            Veg
1  fd1     Veg Creamy Burger            Veg
2  fd2   Cheese Burst Burger            Veg
3  fd3  Paneer Creamy Burger            Veg
4  fd4        Maxican Burger            Veg

In [4]:
# Print the first 5 rows.
Menu.head()

menu_id    r_id      f_id           cuisine  price
0     mn0  567335       fd0  Beverages,Pizzas   40.0
1     mn0  567335  fd669322  Beverages,Pizzas   40.0
2   mn328  158203       fd0         Beverages   65.0
3   mn328  158203  fd669322         Beverages   65.0
4   mn449  158203       fd0         Beverages   65.0

In [5]:
# Print the first 5 rows.
Orders.head()

order_date  sales_qty  sales_amount currency  user_id      r_id
0 2017-10-10        100         41241      INR    49226  567335.0
1 2018-05-08          3            -1      INR    77359  531342.0
2 2018-04-06          1           875      INR     5321  158203.0
3 2018-04-11          1           583      INR    21343  187912.0
4 2018-06-18          6          7176      INR    75378  543530.0

In [6]:
# Print the first 5 rows.
Orders_Type.head()

Order_Id     Type
0  B496840219  Non-Veg
1  C134461623    Other
2  D679770099    Other
3  E622201871      Veg
4  A980473889  Non-Veg

In [7]:
# Print the first 5 rows.
Restaurant.head()

id               name Country    city rating     rating_count  \
0  567335     AB FOODS POINT   India  Abohar     --  Too Few Ratings   
1  531342  Janta Sweet House   India  Abohar    4.4      50+ ratings   
2  158203  theka coffee desi   India  Abohar    3.8     100+ ratings   
3  187912          Singh Hut   India  Abohar    3.7      20+ ratings   
4  543530      GRILL MASTERS   India  Abohar     --  Too Few Ratings   

                      cuisine  \
0            Beverages,Pizzas   
1               Sweets,Bakery   
2                   Beverages   
3            Fast Food,Indian   
4  Italian-American,Fast Food   

                                                link  \
0  https://www.swiggy.com/restaurants/ab-foods-po...   
1  https://www.swiggy.com/restaurants/janta-sweet...   
2  https://www.swiggy.com/restaurants/theka-coffe...   
3  https://www.swiggy.com/restaurants/singh-hut-n...   
4  https://www.swiggy.com/restaurants/grill-maste...   

                                             address  
0  AB FOODS POINT, NEAR RISHI NARANG DENTAL CLINI...  
1  Janta Sweet House, Bazar No.9, Circullar Road,...  
2         theka coffee desi, sahtiya sadan road city  
3    Singh Hut, CIRCULAR ROAD NEAR NEHRU PARK ABOHAR  
4  GRILL MASTERS, ADA Heights, Abohar - Hanumanga...

In [8]:
# Print the first 5 rows.
Users.head()


user_id               name  Age  Gender Marital Status Occupation
0        1    Claire Ferguson   20  Female         Single    Student
1        2     Jennifer Young   24  Female         Single    Student
2        3  Jermaine Roberson   22    Male         Single    Student
3        4   Rachel Carpenter   22  Female         Single    Student
4        5       Shawn Parker   22    Male         Single    Student

In [9]:
# Analyse the Datatypes of Each Table.
Datasets = [Food, Menu, Orders, Orders_Type, Restaurant, Users]
for i in Datasets:
    print(i.dtypes, "\n")
#All Datatypes are Correct.

f_id              object
item              object
veg_or_non_veg    object
dtype: object 

menu_id     object
r_id         int64
f_id        object
cuisine     object
price      float64
dtype: object 

order_date      datetime64[ns]
sales_qty                int64
sales_amount             int64
currency                object
user_id                  int64
r_id                   float64
dtype: object 

Order_Id    object
Type        object
dtype: object 

id               int64
name            object
Country         object
city            object
rating          object
rating_count    object
cuisine         object
link            object
address         object
dtype: object 

user_id            int64
name              object
Age                int64
Gender            object
Marital Status    object
Occupation        object
dtype: object 



### Check and Handle the Null Values

In [10]:
# Check the Null Values from the Food Table.
Food.isnull().sum()


f_id              8
item              8
veg_or_non_veg    8
dtype: int64

In [11]:
# There are 8 Null Values in Each Column of Food Table.
# All Null Values might Belong to the Same Row.

In [12]:
# Print the Null Values.
Food[Food["f_id"].isnull()]

f_id item veg_or_non_veg
13540   NaN  NaN            NaN
99966   NaN  NaN            NaN
99988   NaN  NaN            NaN
99989   NaN  NaN            NaN
99991   NaN  NaN            NaN
99993   NaN  NaN            NaN
99994   NaN  NaN            NaN
166711  NaN  NaN            NaN

In [14]:
# Remove the Null Values from the Food Table.
Food.dropna(inplace=True)

In [15]:
# Check if the null values have been Deleted Correctly or Not.
Food.isnull().sum()

f_id              0
item              0
veg_or_non_veg    0
dtype: int64

In [16]:
# Check the Null Values from Menu Table.
Menu.isnull().sum()

menu_id    0
r_id       0
f_id       0
cuisine    0
price      1
dtype: int64

In [17]:
# There are only 1 Null Value in "Price" Column.
# Print the row containing Null values.
Menu[Menu["price"].isnull()]

menu_id    r_id      f_id       cuisine  price
878051  mn414862  496987  fd413746  South Indian    NaN

In [18]:
# Remove the Null Value.
Menu.dropna(inplace=True)

In [19]:
# Check if the null values have been Deleted Correctly or Not.
Menu.isnull().sum().any()

np.False_

In [20]:
# Check the Null Values from the Orders Table.
Orders.isnull().sum()

order_date         0
sales_qty          0
sales_amount       0
currency           0
user_id            0
r_id            1617
dtype: int64

In [21]:
# Count the Null Values.
Cnt_Null_Orders = Orders["r_id"].isnull().sum()
Cnt_Null_Orders


np.int64(1617)

In [22]:
# Calculate the Percentage of Null Values.
Null_Value_Per = (Cnt_Null_Orders / len(Orders) ) * 100
Null_Value_Per

np.float64(1.0759843227021382)

In [23]:
# Print the Rows That have Null Values.
Orders[Orders["r_id"].isnull()].head()

order_date  sales_qty  sales_amount currency  user_id  r_id
148664 2019-03-19          1          1625      INR    50862   NaN
148665 2019-03-19          1          1741      INR    25120   NaN
148666 2019-03-20          1          1463      INR    84260   NaN
148667 2019-03-20          1          2949      INR    14329   NaN
148668 2019-03-21          1          1241      INR    38858   NaN

In [24]:
# Analyse the Null Values in Restaurant Table.
Restaurant.isnull().sum()

id               0
name            86
Country          0
city             0
rating          86
rating_count    86
cuisine         99
link             0
address         86
dtype: int64

In [25]:
# Print the Null Values
Restaurant[Restaurant["name"].isnull()].head()

id name Country                       city rating rating_count  \
31044  397961  NaN   India  Greater Mohali,Chandigarh    NaN          NaN   
32912  308071  NaN   India        West Chd,Chandigarh    NaN          NaN   
33046  308662  NaN   India        West Chd,Chandigarh    NaN          NaN   
33488  170889  NaN   India       South Chd,Chandigarh    NaN          NaN   
36492  390394  NaN   India        George Town,Chennai    NaN          NaN   

      cuisine                                               link address  
31044     NaN  https://www.swiggy.com/restaurants/5-tara-grea...     NaN  
32912     NaN  https://www.swiggy.com/restaurants/food-under-...     NaN  
33046     NaN  https://www.swiggy.com/restaurants/franks-frie...     NaN  
33488     NaN  https://www.swiggy.com/restaurants/dhaba-7-pre...     NaN  
36492     NaN  https://www.swiggy.com/restaurants/frozen-cafe...     NaN

In [26]:
# Analyze whether the Null values in the Restaurant Table and the Null Values in the Orders' R_id Column have a relationship or not.

In [27]:
# Get the Id's of Restaurant that Have Null values
Null_Restaurant_id = Restaurant[Restaurant["name"].isnull()]["id"]
Null_Restaurant_id

31044     397961
32912     308071
33046     308662
33488     170889
36492     390394
           ...  
137121    256827
137545    567595
137613    116615
138379    401760
141567    463812
Name: id, Length: 86, dtype: int64

In [28]:
#Count of Null Values
Count_of_Null_Rest = len(Null_Restaurant_id)
Count_of_Null_Rest

86

In [29]:
# Get the Details of the Orders Table where the Restaurant IDs have Null values.
Orders[Orders["r_id"].isin(Null_Restaurant_id)].head()

order_date  sales_qty  sales_amount currency  user_id      r_id
31044 2019-11-12          1            79      INR    76969  397961.0
32912 2020-01-13          1            65      INR    66468  308071.0
33046 2020-05-28          2           431      INR    81350  308662.0
33488 2018-05-02          2           463      INR    76025  170889.0
36492 2018-11-16          1            46      INR    49014  390394.0

There is No relationship Betweent the Null Values of R_Id in Orders Table and Restaurant Table

In [30]:
# Calculat the Percentage of Null Values in Restaurant Table
Cnt_Null_Percent = (Count_of_Null_Rest / len(Restaurant) ) * 100
Cnt_Null_Percent

0.05789686279789955

Null Values are 0.05 Percent of total Restaurant Values
We Can Drop these Null Values

In [31]:
# Remove The Null Values from the Restaurant Dataset
Restaurant.dropna(subset="name",inplace=True)

In [32]:
# After Droping Null Values, Count the Entries in The Restaurant Table
len(Restaurant)

148454

Before there was 148540 Rows
Now it is 148454 Rows


In [33]:
# Re-Analyse the Null Values in Restaurant Table
Restaurant.isnull().sum()
# Still Cuisine Have 13 Null Values

id               0
name             0
Country          0
city             0
rating           0
rating_count     0
cuisine         13
link             0
address          0
dtype: int64

In [34]:
# Print the Null Values in Restaurant's "Cuisine" Column
Restaurant[Restaurant["cuisine"].isnull()].head()

id                         name Country                     city  \
47768  155151  Pocket Plates By Abongchiiz   India          GTB Nagar,Delhi   
51698  245476    NEW YORK WAFFLES & DINGES   India  Greater Kailash 2,Delhi   
52218   62718          34 Chowringhee Lane   India        Ashok Vihar,Delhi   
52515  349965     TRP-Tandoor Roll Paratha   India    South Extension,Delhi   
57165  319007             Kathi Roll Point   India       Lajpat Nagar,Delhi   

      rating     rating_count cuisine  \
47768     --  Too Few Ratings     NaN   
51698     --  Too Few Ratings     NaN   
52218    3.6     100+ ratings     NaN   
52515     --  Too Few Ratings     NaN   
57165     --  Too Few Ratings     NaN   

                                                    link  \
47768  https://www.swiggy.com/restaurants/pocket-plat...   
51698  https://www.swiggy.com/restaurants/new-york-wa...   
52218  https://www.swiggy.com/restaurants/34-chowring...   
52515  https://www.swiggy.com/restaurants/trp-tandoor...   
57165  https://www.swiggy.com/restaurants/kathi-roll-...   

                                                 address  
47768  Pocket Plates By Abongchiiz, SHOP NO-15 DDA MA...  
51698  NEW YORK WAFFLES & DINGES, A 6 KAILASH COLONY,...  
52218  34 Chowringhee Lane, Shop no. 2, Phase 1, J bl...  
52515  TRP-Tandoor Roll Paratha, D-59, Panchsheel Enc...  
57165  Kathi Roll Point, SHOP NO.42,NEHRU NAGAR NEW D...

In [35]:
# Analyse the Number of Unique Values in "Cuisine" Column
Restaurant["cuisine"].nunique()

2132

In [36]:
# Print the Most Repeated "Cuisine" Value using Mode Function.
Most_Repeated_Cuisine = Restaurant["cuisine"].mode()
Most_Repeated_Cuisine
# Most Repeated Cuisine is "North Indian & Chinese"

0    North Indian,Chinese
Name: cuisine, dtype: object

In [37]:
# Analyse the Number of Values which Have Most Repeated Cuisine.
len(Restaurant[Restaurant["cuisine"].isin(Most_Repeated_Cuisine ) ])

6471

Almost 6471 Restaurant have "North Indian & Chinese" Cuisine

In [39]:
# impute the Null Values with "North Indian & Chinese"
Restaurant["cuisine"] = Restaurant["cuisine"].fillna(
    Restaurant["cuisine"].mode()[0]
)

In [40]:
# Check one of the Filled Value
Restaurant[Restaurant["id"] == 155151]

id                         name Country             city rating  \
47768  155151  Pocket Plates By Abongchiiz   India  GTB Nagar,Delhi     --   

          rating_count               cuisine  \
47768  Too Few Ratings  North Indian,Chinese   

                                                    link  \
47768  https://www.swiggy.com/restaurants/pocket-plat...   

                                                 address  
47768  Pocket Plates By Abongchiiz, SHOP NO-15 DDA MA...

In [41]:
# Re-Analuse the Null Values in Restaurant Table
Restaurant.isnull().sum()

id              0
name            0
Country         0
city            0
rating          0
rating_count    0
cuisine         0
link            0
address         0
dtype: int64

*Now Restaurant Table have No Null Values*

In [43]:
# Analyse the Null Values in Orders_Type Table
Orders_Type.isnull().sum()

Order_Id    0
Type        0
dtype: int64

*No Null Values in the Orders Table*

In [45]:
# Analyse the Null Values in Users Table
Users.isnull().sum()

user_id           0
name              0
Age               0
Gender            0
Marital Status    0
Occupation        0
dtype: int64

*No Null Values in the Users Table*

## Formating the Tables

 **Rename Columns**

In [49]:
# Print the Columns of Food Table
Food.columns

Index(['f_id', 'item', 'veg_or_non_veg'], dtype='object')

In [50]:
# Rename the Columns in Food Table.
Food.rename(columns={"f_id":"Food_id","veg_or_non_veg":"Food_Type","item":"Item"},inplace=True)
Food.columns

Index(['Food_id', 'Item', 'Food_Type'], dtype='object')

In [51]:
# Print the Columns of Menu Table
Menu.columns

Index(['menu_id', 'r_id', 'f_id', 'cuisine', 'price'], dtype='object')

In [52]:
# Rename the Columns in Menu Table.
Menu.rename(columns={"menu_id":"Menu_id","r_id":"Restaurant_id",
                     "f_id":"Food_id","cuisine":"Cuisine","price":"Price"}, inplace=True)
Menu.columns # Print New Columns's Name

Index(['Menu_id', 'Restaurant_id', 'Food_id', 'Cuisine', 'Price'], dtype='object')

In [53]:
# Print the Columns of Orders Table
Orders.columns

Index(['order_date', 'sales_qty', 'sales_amount', 'currency', 'user_id',
       'r_id'],
      dtype='object')

In [54]:
# Rename the Columns in Orders Table.
Orders.rename(columns={"order_Date":"Order_date","sales_qty":"Sales_QTY","sales_amount":"Sales_amount",
                       "Currency":"Currency","user_id":"User_id","r_id":"Restaurant_id"}, inplace=True)
Orders.columns # Print New Columns's Name

Index(['order_date', 'Sales_QTY', 'Sales_amount', 'currency', 'User_id',
       'Restaurant_id'],
      dtype='object')

In [55]:
# Print the Columns of Restaurant Table
Restaurant.columns

Index(['id', 'name', 'Country', 'city', 'rating', 'rating_count', 'cuisine',
       'link', 'address'],
      dtype='object')

In [56]:
# Rename the Columns in Restaurant Table.
Restaurant.rename(columns={"id":"Restaurant_id","name":"Name","city":"City",
                           "rating":"Rating","rating_count":"Rating_count","cuisine":"Cuisine","link":"Link","address":"Address"},
                           inplace=True)
Restaurant.columns # Print New Columns's Name

Index(['Restaurant_id', 'Name', 'Country', 'City', 'Rating', 'Rating_count',
       'Cuisine', 'Link', 'Address'],
      dtype='object')

In [57]:
# Print the Columns of Users Table
Users.columns

Index(['user_id', 'name', 'Age', 'Gender', 'Marital Status', 'Occupation'], dtype='object')

In [58]:
# Rename the Columns in Restaurant Table.
Users.rename(columns={"user_id":"User_id","name":"Name"}, inplace=True)
Users.columns # Print New Columns's Name

Index(['User_id', 'Name', 'Age', 'Gender', 'Marital Status', 'Occupation'], dtype='object')

*Drop Unnecessary Columns and Tables.*

In [59]:
# Drop the Un_neccesary column from the restaurant table
Restaurant.drop(columns=["Link","Address"], axis=1, inplace=True)

In [60]:
Restaurant.head(2)

Restaurant_id               Name Country    City Rating     Rating_count  \
0         567335     AB FOODS POINT   India  Abohar     --  Too Few Ratings   
1         531342  Janta Sweet House   India  Abohar    4.4      50+ ratings   

            Cuisine  
0  Beverages,Pizzas  
1     Sweets,Bakery

In [61]:
# Delete the Orders_type Table 's DataFrame
del Orders_Type

In [63]:
# Delete the Orders_Type file from the system
import os
path = r"D:\DATA ANALYST\Swiggy\Raw Dataset\orders_type.xlsx"
try:
    os.remove(path)
except Exception as e:
    print(e) 

## Store the Dataset into the System

In [66]:
# Store the Cleaned and Processed Data into the System.
Food.to_csv(r"D:\DATA ANALYST\Swiggy\Processed Data (Python)\Food.csv", index=False)
Menu.to_csv(r"D:\DATA ANALYST\Swiggy\Processed Data (Python)\Menu.csv", index=False)
Orders.to_csv(r"D:\DATA ANALYST\Swiggy\Processed Data (Python)\Orders.csv", index=False)
Restaurant.to_csv(r"D:\DATA ANALYST\Swiggy\Processed Data (Python)\Restaurant.csv", index=False)
Users.to_csv(r"D:\DATA ANALYST\Swiggy\Processed Data (Python)\Users.csv", index=False)


## Thank You